src/io/asyncsocket.c

```diff
@@ -497,6 +515,102 @@ static void push_name_and_port(MVMThreadContext *tc, struct sockaddr_storage *na
     MVM_repr_push_o(tc, arr, port_o);
 }
 
+typedef struct {
+    uv_tcp_t         *socket;
+    MVMThreadContext *tc;
+    int               work_idx;
+} SocketInfo;
+
+static void socket_setup(MVMThreadContext *tc, uv_loop_t *loop, MVMObject *async_task, void *data) {
+    int r;
+
+    /* Add to work in progress. */
+    MVMObject    *arr = MVM_repr_alloc_init(tc, tc->instance->boot_types.BOOTArray);
+    MVMAsyncTask *t   = (MVMAsyncTask *)async_task;
+    SocketInfo   *si  = (SocketInfo *)data;
+    si->tc            = tc;
+    si->work_idx      = MVM_io_eventloop_add_active_work(tc, async_task);
+
+    /* Create and initialize socket. */
+    si->socket       = MVM_malloc(sizeof(uv_tcp_t));
+    si->socket->data = data;
+    if ((r = uv_tcp_init(loop, si->socket)) < 0) {
+        MVMROOT2(tc, arr, t, {
+            MVM_repr_push_o(tc, arr, t->body.schedulee);
+            MVM_repr_push_o(tc, arr, tc->instance->boot_types.BOOTIO);
+            MVMString *msg_str = MVM_string_ascii_decode_nt(tc, tc->instance->VMString, uv_strerror(r));
```

Comment:
ISO C90 forbids mixed declarations and code

Reply:
Should be fixed now